### PR TITLE
refactor metadata for template

### PR DESCRIPTION
**Pull request by AngryBerryMS**

### CODE REVIEW

File: src/docfx/data/docs/metadata.json [hunk 1]
```diff
@@ -29,59 +29,6 @@
             "type": "boolean",
             "description": "Indicate whether the depot need to API Scan"
         },
-        "product": {
-            "type": ["array", "null"],
```
Comment (yufeih) [inline src/docfx/data/docs/metadata.json:33]: only remove `htmlMetaName`. Other properties like `type` is still used for metadata validation

File: src/docfx/build/metadata/MetadataProvider.cs [hunk 2]
```diff
@@ -19,27 +18,60 @@ internal class MetadataProvider
 
         private readonly ConcurrentDictionary<FilePath, Watch<(ErrorList, UserMetadata)>> _metadataCache = new();
 
-        public ICollection<string> HtmlMetaHidden { get; }
-
-        public IReadOnlyDictionary<string, string> HtmlMetaNames { get; }
-
-        public MetadataProvider(Config config, Input input, BuildScope buildScope, JsonSchemaLoader jsonSchemaLoader)
+        public ICollection<string> HtmlMetaHidden { get; } = new List<string>()
```
Comment (yufeih) [inline src/docfx/build/metadata/MetadataProvider.cs:21]: Move these two properties to `HtmlUtility.cs`, they are only used in `CreateHtmlMetaTags` method.

Comment (yufeih) [inline src/docfx/build/metadata/MetadataProvider.cs:21]:
```suggestion
        public IReadOnlyList<string> HtmlMetaHidden { get; } = new[]
```

Comment (AngryBerryMS) [reply src/docfx/build/metadata/MetadataProvider.cs]: Done.

File: src/docfx/data/docs/metadata.json (hunk 1 above)
Comment (AngryBerryMS) [reply src/docfx/data/docs/metadata.json]: Oh sorry. Fixed.

File: src/docfx/lib/HtmlUtility.cs [hunk 3]
```diff
@@ -18,6 +18,54 @@ internal static class HtmlUtility
...
+            "is_dynamic_rendering",
+            "need_preview_pull_request",
+            "moniker_type",
+            "is_significant_update",
+            "serviceData",
+            "github_contributors",
+            "is_hidden",
+        };
+
+        public static readonly Dictionary<string, string> HtmlMetaNames = new()
```
Comment (live1206) [inline src/docfx/lib/HtmlUtility.cs:58]: nit: IReadOnlyDictionary

File: src/docfx/lib/HtmlUtility.cs [hunk 4]
```diff
@@ -18,6 +18,54 @@ internal static class HtmlUtility
     {
         public delegate void TransformHtmlDelegate(ref HtmlReader reader, ref HtmlWriter writer, ref HtmlToken token);
 
+        public static readonly string[] HtmlMetaHidden = new[]
```
Comment (yufeih) [inline src/docfx/lib/HtmlUtility.cs:21]: This doesn't need to be public.
```suggestion
        private static readonly string[] s_htmlMetaHidden = new[]
```

File: src/docfx/lib/HtmlUtility.cs (hunk 3 above)
Comment (AngryBerryMS) [reply src/docfx/lib/HtmlUtility.cs:58]: Want to be compatible  with https://github.com/dotnet/docfx/blob/2146708759fad1b02b5f6e2bac48e202efecb8e6/src/docfx/lib/HtmlUtility.cs#L39

**Review by yufeih — APPROVED** [review]
also remove the properties from JsonSchema.cs

**Pull request merged**